Doc: Single DC Multipod L3LS Update (#6662)
aristanetworks/ansible-avd: ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/README.md

This example only teaches some aspects of an L3LS EVPN/VXLAN build; please see the [single DC L3LS example](../single-dc-l3ls/README.md) for additional information. It supplements the single DC example, concentrating on the unique elements of multiple PODs/5-stage Clos.

## AVD Playground

--8<--
ansible_collections/arista/avd/examples/common/start-avd-playground.md
--8<--

## Installation

--8<--
ansible_collections/arista/avd/examples/common/example-installation.md
--8<--

```shell
ansible-avd-examples/ (or wherever the playbook was run)
  |── single-dc-multipod-l3ls
    ├── ansible.cfg
    ├── build.yml
    ├── clab
    ├── deploy-cvp.yml
    ├── deploy.yml
    ├── documentation
    ├── group_vars
    ├── images
    ├── intended
    ├── inventory.yml
    ├── README.md
    └── validate.yml
```

## Overall design overview

### Physical topology

The drawing below shows the physical topology used in this example. The interface assignments shown here are referenced across the entire example, so keep that in mind if this example must be adapted to a different topology.

> ![Figure: Arista Leaf Spine physical topology](images/l3ls-multipod.svg)

Reviewer: dc1-leaf2b has extra/unused port 6

Author: Great find, thank you

### Fabric design

The fabric is a basic L3LS EVPN/VXLAN design with a multi-pod (5-stage Clos) architecture.

## Ansible inventory, group vars, and naming scheme

The following drawing shows a graphic overview of the Ansible inventory, group variables, and naming scheme used in this example:

![Figure: Arista Leaf Spine physical topology](images/inventory.svg)

> The SUPERSPINES group has been added, as well as POD1 and POD2 groups with PODX_LEAFS and PODX_SPINES under each. The EVPN_SERVICES and ENDPOINT_CONNECT allow separation of YAML files, and putting the PODX_LEAFS under them will build the appropriate configs for those devices (VXLAN/VLAN/anycast gateways do not get instantiated on spines, of course).

Reviewer: ```suggestion
The SUPERSPINES group has been added, as well as POD1 and POD2 groups with PODX_LEAFS and PODX_SPINES under each. The EVPN_SERVICES and CONNECTED_ENDPOINTS allow separation of YAML files, and putting the PODX_LEAFS under them will build the appropriate configs for those devices (VXLAN/VLAN/anycast gateways do not get instantiated on spines, of course).
```

### Content of the inventory.yml file

```yaml title="inventory.yml"
--8<--
ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/inventory.yml
--8<--
```

## Fabric files

With the topology, the following YAML files are used in group_vars:

- FABRIC/fabric_ansible_connectivity.yml
- FABRIC/fabric_variables.yml
- SUPERSPINES/superspines.yml
- POD1/pod1.yml
- POD2/pod2.yml
- EVPN_SERVICES/evpn_services.yml

> - ENDPOINT_CONNECT/endpoints.yml

Reviewer: ```suggestion
- CONNECTED_ENDPOINTS/endpoints.yml
```

The leaf configurations, EVPN services, and endpoints sections aren't affected by the multi-pod format.

=== "POD1"
    The POD1 and POD2 YAML files contain the descriptions of the leafs and spines. Note that each POD's spines have its own unique ASN (eBGP). Also, the spines now have uplink interfaces and uplink switches specified (to the superspines) with the `uplink_switches` and `uplink_switch_interfaces` directives. The uplink pool can overlap between the PODs in a DC. If doing multi-DC, the pools should be on different subnets.

    ```yaml title="POD1/pod1.yml"
    --8<--
    ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/POD1/pod1.yml
    --8<--
    ```

=== "POD2"
    Please note the similarities between POD1 and POD2.

    ```yaml title="POD2/pod2.yml"
    --8<--
    ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/POD2/pod2.yml
    --8<--
    ```

## Endpoint connectivity

> The final group variables file provides an example of connecting two servers across a leaf pair.

Reviewer: Should `EVPN_SERVICES/evpn_services.yml` be covered as well?

Reviewer 2: I'd leave this up to Julio as well. In my opinion README is very optional part of the example inventory and it was quite clear in it's current form. We have no obligation to cover every single data structure there. Especially now that we have AVD playground for every example inventory.

Author: Right, covering again would be a repeat of the previous example that this builds from (single DC) arguably we could also remove this section as well

> ```yaml title="ENDPOINT_CONNECT/endpoints.yml"

Reviewer: ```suggestion
```yaml title="CONNECTED_ENDPOINTS/endpoints.yml"
```

--8<--

> ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/ENDPOINT_CONNECT/endpoints.yml

Reviewer: ```suggestion
ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/CONNECTED_ENDPOINTS/endpoints.yml
```

---
# This title is used for search results

> title: AVD example for a single data center using multiple PODs for L3LS

Reviewer 3: The title should match other examples 
```suggestion
title: Mutli-Pod L3LS
```

---
<!--
  ~ Copyright (c) 2023-2026 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

> # AVD example for a single data center with Multi-Pod using L3LS

Reviewer 2: Can we use the "5-stage" as the primary name everywhere instead of Multi-Pod? Not insisting, but I think it's the term most people are used to.

## Introduction

This example shows how to create a multi-pod environment (a 5-stage Clos) in a single DC environment. Of course, this can be used in multiple DCs, but this example is only for two PODs in a single DC.

> Also included is an example of connecting an external router to a VRF/tenant.

Reviewer 2: why? =)
let's keep it clean and focused on a single concept at a time only

Author: Fair enough, ill remove the external router portion

> This example only teaches some aspects of an L3LS EVPN/VXLAN build; please see the [single DC L3LS example](../single-dc-l3ls/README.md) for additional information. It supplements the single DC example, concentrating on the unique elements of multiple PODs/5-stage Clos.

Reviewer 2: I think it's great to state that example is not a complete prod inventory and is focused on 5-stage specifics. However I'd avoid explicit references and links to other examples and keep them all independent.

Author: Internal references should be encouraged, great for readers that want more context and can stay within our documentation.

Reviewer 2: Disagree for examples. =) But it is certainly not critical. Let's discuss this at a later stage.

Ansible playbooks are included to show the following:

- Building the intended configuration and documentation
- Deploying the configuration via CloudVision or directly to the switches via eAPI

> - Validating the configuration

Reviewer 2: I'd cut this

    ```yaml title="POD1/pod1.yml"
    --8<--
    ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/POD1/pod1.yml
    --8<--
    ```

=== "POD2"
    Please note the similarities between POD1 and POD2.

    ```yaml title="POD2/pod2.yml"
    --8<--
    ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/POD2/pod2.yml
    --8<--
    ```

## Connecting an External Router

In addition to the multi-pod, this example has a tenant/VRF connecting to an external network via a router (R1). This is defined in the `evpn_services.yml` file. The `l3_interfaces` parameter creates an L3 interface in the VRF on a specific leaf and the `bgp_peer` section defines the BGP peering with the external router.

```yaml title="EVPN_SERVICES/evpn_services.yml"
--8<--
ansible_collections/arista/avd/examples/single-dc-multipod-l3ls/group_vars/EVPN_SERVICES/evpn_services.yml
--8<--

> ```

Reviewer 2: remove
